sign/dilithium/mode3/internal/sample.go

```diff
@@ -122,8 +305,32 @@ func PolyDeriveUniformLeqEta(p *common.Poly, seed *[32]byte, nonce uint16) {
 // v[i] will not be normalized, but have coefficients in the
 // interval (q-γ₁,q+γ₁).
 func VecLDeriveUniformLeGamma1(v *VecL, seed *[48]byte, nonce uint16) {
-	for i := 0; i < L; i++ {
-		PolyDeriveUniformLeGamma1(&v[i], seed, nonce+uint16(i))
+	if !DeriveX4Available {
+		for i := 0; i < L; i++ {
+			PolyDeriveUniformLeGamma1(&v[i], seed, nonce+uint16(i))
+		}
+		return
+	}
+
+	// PolyDeriveUniformLeGamma1X4 is slower than, but not twice as slow as,
+	// PolyDeriveUniformLeGamma.
+	if L == 2 {
```

Comment:
You can construct the input vector iterating in a for-loop with L as bound. The non-assigned values are taken by default to 0 and nil.

Reply:
Ah, good trick!